validator/client/validator.go

```diff
@@ -236,8 +245,34 @@ func (v *validator) WaitForActivation(ctx context.Context) error {
 
 	validatingKeys, err := v.keyManager.FetchValidatingPublicKeys(ctx)
 	if err != nil {
-		return errors.Wrap(err, "could not fetch validating keys")
+		return errors.Wrap(err, msgCouldNotFetchKeys)
 	}
+	if len(validatingKeys) == 0 {
+		log.Warn(msgNoKeysFetched)
+
+		ticker := time.NewTicker(keyRefetchPeriod)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				keys, err := v.keyManager.FetchValidatingPublicKeys(ctx)
+				if err != nil {
+					return errors.Wrap(err, msgCouldNotFetchKeys)
+				}
+				if len(keys) == 0 {
+					log.Warn(msgNoKeysFetched)
+					continue
+				}
+				validatingKeys = keys
+				break
+			case <-ctx.Done():
+				log.Debug("Context closed, exiting fetching validating keys")
+				return errors.New("context closed, no longer attempting to fetch validating keys")
```

Comment:
```suggestion
				return ctx.Err()
```

I like `ctx.Err()` better here. Current error message sounds duplicated with the debug log